Sources/SourceParsingFramework/Utilities/Logger.swift

```diff
@@ -67,49 +77,61 @@ public func set(minLoggingOutputLevel level: LoggingLevel) {
 /// Log the given message at the `debug` level.
 ///
 /// - parameter message: The message to log.
+/// - parameter path: The file path this error applies to
+///   (if specified, the log output changes to one that matches compiler error and can be parsed easily by build systems)
 /// - note: The mesasge is only logged if the current `minLoggingOutputLevel`
 /// is set at or below the `debug` level.
-public func debug(_ message: String) {
-    log(message, atLevel: .debug)
+public func debug(_ message: String, path: String? = nil) {
+    log(message, atLevel: .debug, path: path)
 }
 
 /// Log the given message at the `info` level.
 ///
 /// - parameter message: The message to log.
+/// - parameter path: The file path this error applies to
+///   (if specified, the log output changes to one that matches compiler error and can be parsed easily by build systems)
 /// - note: The mesasge is only logged if the current `minLoggingOutputLevel`
 /// is set at or below the `info` level.
-public func info(_ message: String) {
-    log(message, atLevel: .info)
+public func info(_ message: String, path: String? = nil) {
+    log(message, atLevel: .info, path: path)
 }
 
 /// Log the given message at the `warning` level.
 ///
 /// - parameter message: The message to log.
+/// - parameter path: The file path this error applies to
+///   (if specified, the log output changes to one that matches compiler error and can be parsed easily by build systems)
 /// - note: The mesasge is only logged if the current `minLoggingOutputLevel`
 /// is set at or below the `warning` level.
-public func warning(_ message: String) {
-    log(message, atLevel: .warning)
+public func warning(_ message: String, path: String? = nil) {
+    log(message, atLevel: .warning, path: path)
 }
 
 /// Log the given message at the `error` level and terminate.
 ///
 /// - parameter message: The message to log.
+/// - parameter path: The file path this error applies to
+///   (if specified, the log output changes to one that matches compiler error and can be parsed easily by build systems)
 /// - returns: `Never`.
 /// - note: The mesasge is only logged if the current `minLoggingOutputLevel`
 /// is set at or below the `error` level.
 /// - note: This method terminates the program. It returns `Never`.
-public func error(_ message: String) -> Never {
-    log(message, atLevel: .error)
+public func error(_ message: String, path: String? = nil) -> Never {
+    log(message, atLevel: .error, path: path)
     exit(1)
 }
 
-private func log(_ message: String, atLevel level: LoggingLevel) {
+private func log(_ message: String, atLevel level: LoggingLevel, path: String?) {
     #if DEBUG
         UnitTestLogger.instance.log(message, at: level)
     #endif
 
     if level.rawValue >= minLoggingOutputLevel.value {
-        print("\(level.emoticon) \(message)")
+        if let path = path {
+            print("\(path):1:1: \(level.text): \(message)")
```

Comment:
is :1:1 supposed to be a line range?  is it going to be passed in as arg?

Reply:
yes, it's line number and then the column.  for now, I'm leaving it out as our tools don't really have that level of detail (also, clicking on the needle error and jumping to the file is already pretty cool -- even though it takes you to the top of the file)

Reply:
nice